### DISCUSSION TITLE
WakaTime custom period

### DISCUSSION THREAD

**Discussion by wleoncio** [anuraghazra/github-readme-stats#702] [Q&A] [answered]
The [Wakatime stats card](https://github.com/anuraghazra/github-readme-stats#wakatime-week-stats) seems to only be able to show stats for the last 7 days. Is there a query parameter to customize this? I couldn't find anything about it on the [readme](https://github.com/anuraghazra/github-readme-stats/blob/master/readme.md) file or on the Issues/Discussions pages.

**Comment by haaarshsingh** [comment, marked as answer]
To my understanding no, you cannot have it go further. It's based upon the stats from the [wakatime profile] (I'll link mines just as an example) it only has weekly data for your coding activity. 

[wakatime profile]: https://wakatime.com/@harshhhdev

**Reply by anuraghazra** [reply]
https://wakatime.com/developers#stats

**Comment by nyufeng** [comment]
根据我的测试，如果想要通过 API 获取超过 7 天的数据，则需要鉴权，密钥将会暴露。
可以通过 share embed 来折中实现
我已经有一个例子 #709 
https://wakatime.com/share/embed
https://vercel.ddnpc.com/api/wakatime-embed?username=instrye&embedCode=2916255c-a02d-4b4b-b7e8-d3533f0fd393

According to my test, if you want to obtain data for more than 7 days through the API, authentication is required and the key will be exposed.
Can be achieved through share embed
I already have an example #709
https://wakatime.com/share/embed
https://vercel.ddnpc.com/api/wakatime-embed?username=instrye&embedCode=2916255c-a02d-4b4b-b7e8-d3533f0fd393
